### ISSUE TITLE
Suggest usage of `ref` instead of `&*`

### ISSUE THREAD

**Issue by killercup** [rust-lang/rust-clippy#188]
Like I commented in #170, I have code where a closure takes only one references value.

There are two ways of writing it:

``` rust
.and_then(|dir| search(&*dir, file).map_err(From::from))
```

which is a bit weird, or

``` rust
.and_then(|ref dir| search(dir, file).map_err(From::from)))
```

which in my mind is a bit clearer. The latter should be suggested by clippy.

I'm not sure clippy is the right place to put this (it's not really an error), but it could help beginners write better code.

**Comment by Manishearth** [comment]
Might want to keep this an `Allow` only lint or something, but I like this idea.
